### ISSUE TITLE
bug in 'quotient()' over integers?

### ISSUE THREAD

**Issue by jakobkroeker** [Macaulay2/M2#203]
the following example fails (at least in M2 1.6)
```
R = ZZ[x,y,z]

I = ideal(17*x*z+10*x-43,41*x*y-16*x,47*z^2-2*z)
J = ideal( 34*x-16,-2*y*z,-20)

Q = quotient(J, I)
gQ = quotient(ideal gens gb J,ideal gens gb I)
assert (Q == gQ)
```
output
```
i1 : R = ZZ[x,y,z]

o1 = R

o1 : PolynomialRing

i2 : 
     I = ideal(17*x*z+10*x-43,41*x*y-16*x,47*z^2-2*z)

                                              2
o2 = ideal (17x*z + 10x - 43, 41x*y - 16x, 47z  - 2z)

o2 : Ideal of R

i3 : J = ideal( 34*x-16,-2*y*z,-20)

o3 = ideal (34x - 16, -2y*z, -20)

o3 : Ideal of R

i4 : 
     Q = quotient(J, I)

o4 = ideal (20, 2x - 8)

o4 : Ideal of R

i5 : gQ = quotient(ideal gens gb J,ideal gens gb I)

o5 = ideal (20, 2x - 8, 2y*z)

o5 : Ideal of R

i6 : assert (Q == gQ)

i7 : stdio:8:1:(3): error: assertion failed
```

**Comment by jakobkroeker** [comment]
@mikestillman 

could you provide a workaround for  trim in the meanwhile 
before a fix is available?

**Comment by mikestillman** [comment]
Fixed in commit 172cdeff2214f639586facd5a59455df8b57c5db.

The logic for knowing when all original generators have been considered (during a GB computation, to find a small set of generators, called incorrectly "mingens" in some cases) was incorrect over ZZ in some instances.  The code has been changed so the logic is more transparent (and correct!).
